subsys/bluetooth/mesh/pb_adv.c

```diff
@@ -460,7 +460,7 @@ static void gen_prov_start(struct prov_rx *rx, struct net_buf_simple *buf)
 	link.rx.last_seg = START_LAST_SEG(rx->gpc);
 
 	if ((link.rx.seg & BIT(0)) &&
-	    (find_msb_set(~link.rx.seg) >= link.rx.last_seg)) {
```

Comment:
`find_msb_set(~link.rx.seg) `this always return **32** here.

Reply:
Replace `& 0xff` with `& SEG_NVAL`? It's not clear from this line that this is the reason we need 0xff.